tests/cmds/test_wallet.py

```diff
@@ -160,3 +188,854 @@ async def get_coin_records(self, request: GetCoinRecords) -> Dict[str, Any]:
+    # set RPC Client
+    class UpdateDerivationIndexRpcClient(TestWalletRpcClient):
+        async def extend_derivation_index(self, index: int) -> str:
+            self.add_to_log("extend_derivation_index", (index,))
+            return str(index)
+
+    inst_rpc_client = UpdateDerivationIndexRpcClient()  # pylint: disable=no-value-for-parameter
+    test_rpc_clients.wallet_rpc_client = inst_rpc_client
+    index = 600
+    command_args = ["wallet", "update_derivation_index", FINGERPRINT_ARG, "--index", str(index)]
+    success, output = run_cli_command(capsys, root_dir, command_args)
+    assert success
+    # these are various things that should be in the output
+    assert_list = [f"Updated derivation index: {index}"]
+    cli_assert_shortcut(output, assert_list)
+    expected_calls: logType = {
+        "extend_derivation_index": [(index,)],
+    }
+    test_rpc_clients.wallet_rpc_client.check_log(expected_calls)
+
+
+def test_add_token(capsys: object, get_test_cli_clients: Tuple[TestRpcClients, Path]) -> None:
+    test_rpc_clients, root_dir = get_test_cli_clients
+
+    # set RPC Client
+    class AddTokenRpcClient(TestWalletRpcClient):
+        async def create_wallet_for_existing_cat(self, asset_id: bytes) -> Dict[str, int]:
+            self.add_to_log("create_wallet_for_existing_cat", (asset_id,))
+            return {"wallet_id": 3}
+
+        async def set_cat_name(self, wallet_id: int, name: str) -> None:
+            self.add_to_log("set_cat_name", (wallet_id, name))
+            return None  # we don't need to do anything here
+
+    inst_rpc_client = AddTokenRpcClient()  # pylint: disable=no-value-for-parameter
+    test_rpc_clients.wallet_rpc_client = inst_rpc_client
+    command_args = ["wallet", "add_token", FINGERPRINT_ARG, "-nexamplecat"]
+    success, output = run_cli_command(capsys, root_dir, command_args + ["--asset-id", bytes32([1] * 32).hex()])
+    other_success, other_output = run_cli_command(
+        capsys, root_dir, command_args + ["--asset-id", bytes32([3] * 32).hex()]
+    )
+    assert success and other_success
+    # these are various things that should be in the output
+    assert_list = [f"Successfully renamed test1 with wallet_id 2 on key {FINGERPRINT} to examplecat"]
+    other_assert_list = [f"Successfully added examplecat with wallet id 3 on key {FINGERPRINT}"]
+    cli_assert_shortcut(output, assert_list)
+    cli_assert_shortcut(other_output, other_assert_list)
+    expected_calls: logType = {
+        "cat_asset_id_to_name": [(bytes32([1] * 32),), (bytes32([3] * 32),)],
+        "create_wallet_for_existing_cat": [(bytes32([3] * 32),)],
+        "set_cat_name": [(2, "examplecat"), (3, "examplecat")],
+    }
+    test_rpc_clients.wallet_rpc_client.check_log(expected_calls)
+
+
+def test_make_offer(capsys: object, get_test_cli_clients: Tuple[TestRpcClients, Path]) -> None:
+    test_rpc_clients, root_dir = get_test_cli_clients
+
+    # set RPC Client
+    class MakeOfferRpcClient(TestWalletRpcClient):
```

Comment:
I would have expected that some tests could share the mocked client class. Is there no opportunity for reusing this?

Reply:
If the rpc call / function is used in multiple tests it is added to  TestWalletRpcClient directly, so some tests just directly use it instead of making a new class.